pkg/kubernetes/scripts/kube-client-cockpit.js

```diff
@@ -886,7 +895,16 @@
             }
 
             function read() {
-                return runCommand(["kubectl", "config", "view", "--output=json", "--raw"]);
+                var cmd = ["kubectl", "config", "view", "--output=json", "--raw"];
+
+                /* Call kubectl version first incase we have a auth-provider
+                 * that needs to be populated */
+                return runCommand(["kubectl", "version"])
+                        .then(function() {
+                            return runCommand(cmd);
+                        }, function () {
+                            return runCommand(cmd);
```

Comment:
Seems a bit odd to expect `kubectl` to be present and working (for `config`) but be silent about `kubectl version` failing. Should the error be at least logged?

Reply:
If the cluster is down or otherwise unreachable the kubectl version command will fail. If we log it we'll end up with double logs for the same error in those cases once when version tries to connect  and once when we try.